src/apps/lwaftr/lwaftr.lua

```diff
@@ -419,13 +418,16 @@ local function drop_ipv6_packet_from_bad_softwire(lwstate, pkt)
    end
 
    local ipv6_header = get_ethernet_payload(pkt)
-   local ipv6_src_addr = get_ipv6_src_address(ipv6_header)
+   local orig_src_addr = get_ipv6_src_address(ipv6_header)
+   -- If br_addr is specified, use that as the source addr. Otherwise, send it
+   -- back from the IPv6 address it was sent to.
+   local icmpv6_src_addr = br_addr or get_ipv6_dst_address(ipv6_header)
    local icmp_config = {type = constants.icmpv6_dst_unreachable,
                         code = constants.icmpv6_failed_ingress_egress_policy,
                        }
    local b4fail_icmp = icmp.new_icmpv6_packet(
-      lwstate.aftr_mac_b4_side, lwstate.next_hop6_mac, lwstate.aftr_ipv6_ip,
-      ipv6_src_addr, pkt, icmp_config)
+      lwstate.aftr_mac_b4_side, lwstate.next_hop6_mac, icmpv6_src_addr,
+      orig_src_addr --[[now dest--]], pkt, icmp_config)
```

Comment:
I don't like this inline comment here (AFAIK there's no where else in the code an inline comment like this). Do you mind moving it upwards, when `orig_src_addr` is retrieved? Or likewise `icmpv6_src_addr` is obtained from `br_addr or get_ipv6_dst_address`, rename `orig_src_addr` to `dst_addr`.